### PR TITLE
verilator: create package

### PR DESCRIPTION
**Pull request by save-buffer**
Creates verilator package for void linux.

### CODE REVIEW

**Comment by ericonr** [comment]
I believe it's preferrable to add a `pre_configure` step that runs `autoconf` (or `autoreconf`, which is more common) than a whole new build style.

**Comment by save-buffer** [comment]
> I believe it's preferrable to add a `pre_configure` step that runs `autoconf` (or `autoreconf`, which is more common) than a whole new build style.

Oh thanks for telling me! I've updated the PR.

**Comment by save-buffer** [comment]
Also one question I had about the maintainer field: is this the maintainer of the package template or the actual package? I have no relation to verilator personally, I just needed it and thought I'd contribute my template.

**Comment by abenson** [comment]
1. [Squash](https://levelup.gitconnected.com/how-to-squash-git-commits-9a095c1bc1fc) these to a single commit
2. Edit the commit message to follow convention, should be "New package: verilator-4.102"

> is this the maintainer of the package template or the actual package?

The maintainer of the template.

File: srcpkgs/verilator/template [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'foo'
+pkgname=verilator
+version=4.102
+revision=1
+build_style=configure
```
Comment (abenson) [inline srcpkgs/verilator/template:5]: This should be gnu-configure

File: srcpkgs/verilator/template [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+configure_args="--prefix /usr"
```
Comment (abenson) [inline srcpkgs/verilator/template:6]: This can be dropped

File: srcpkgs/verilator/template [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+hostmakedepends="autoconf"
+makedepends="flex"
```
Comment (abenson) [inline srcpkgs/verilator/template:8]: flex should be in hostmakedepends

File: srcpkgs/verilator/template [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+short_desc="C++ Library for RTL emulation"
+maintainer="krassovskysasha@gmail.com"
+license="GPL-3.0"
```
Comment (abenson) [inline srcpkgs/verilator/template:11]: Should be LGPL-3.0-only

File: srcpkgs/verilator/template [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'foo'
+pkgname=verilator
+version=4.102
+revision=1
+build_style=configure
+configure_args="--prefix /usr"
+hostmakedepends="autoconf"
+makedepends="flex"
+short_desc="C++ Library for RTL emulation"
```
Comment (abenson) [inline srcpkgs/verilator/template:9]: Why not use verilator's official description, "SystemVerilog simulator and lint system"

File: srcpkgs/verilator/template [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+maintainer="krassovskysasha@gmail.com"
```
Comment (abenson) [inline srcpkgs/verilator/template:10]: Please use "Name <email>" format

File: srcpkgs/verilator/template [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'foo'
```
Comment (abenson) [inline srcpkgs/verilator/template:1]: `# Template file for 'verilator'`

File: srcpkgs/verilator/template [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+hostmakedepends="autoconf flex"
+short_desc="SystemVerilog simulator and lint system"
+maintainer="Sasha Krassovsky <krassovskysasha@gmail.com>"
+license="LGPL-3.0-only"
+homepage="https://veripool.org/wiki/verilator"
+distfiles="https://github.com/verilator/verilator/archive/v${version}.tar.gz"
+checksum="4e4f4aff00af9a15a61c94b67ed070cd6312ddcc0f0d340a6df2199480064cef"
+
+pre_configure() {
+    autoconf
```
Comment (abenson) [inline srcpkgs/verilator/template:15]: Use tabs.

A good idea is to run `xlint` on the template.

```
% xlint srcpkgs/verilator/template 
srcpkgs/verilator/template:1: Header should be: # Template file for 'verilator'
srcpkgs/verilator/template:15: indent with tabs
```

File: srcpkgs/verilator/template [hunk 9]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'verilator'
+pkgname=verilator
+version=4.102
+revision=1
+build_style=gnu-configure
+hostmakedepends="autoconf flex"
```
Comment (abenson) [inline srcpkgs/verilator/template:6]: I needed to add `libfl-devel` to `makedepends`

**Comment by jnbr** [comment]
Tested locally, builds for all targets and xlint passes.

**Pull request merged**